clustering/marshalling/protostream/pom.xml

```diff
@@ -45,11 +45,31 @@
         Also contains wildfly-clustering-marshalling-spi implementations using ProtoStream.
     </description>
 
-    <properties>
-        <jdk.max.version>9</jdk.max.version>
-        <!-- This property otherwise causes the java8-test execution to crash -->
-        <modular.jdk.args></modular.jdk.args>
-    </properties>
+    <profiles>
+        <profile>
+            <id>disable-tests-on-publish-to-maven-central</id>
+            <activation>
+                <!--
+                     Disabling tests execution during deployment to the maven central repository.
+                     The reasoning for this conditional skip of tests is it is not possible to pass
+                     valid ${modular.jdk.args} value to both NOT MODULAR JDK and MODULAR JDK based surefire execution
+                     that is happening when this subproject is being published to maven central repository
+                     (at least with current approach org.jboss:maven-parent is using).
+                     The precondition for this exclusion is it is expected that this submodule
+                     is properly tested on NOT MODULAR JDK and MODULAR JDKs before tagging.
+                     Note it is still possible to test on both JDK8 & MODULAR JDK
+                     but not on both JDK types at the same time as is the case when ${java8.home} property is set.
+                -->
+                <jdk>[9,)</jdk>
+                <property>
+                    <name>java8.home</name>
+                </property>
+            </activation>
+            <properties>
+                <skipTests>true</skipTests>
+            </properties>
+        </profile>
+    </profiles>
```

Comment:
@pferraro Please approve

Reply:
This would affect clustering/ee/cache/pom.xml as well.

Reply:
Yes, but current situation is it is possible to build & run clustering subsystem only on JDK15 and below.
If one tries to use JDK16 and above it fails.
This PR moves it one step forward i.e. it would be possible to build & run it on both JDK15- & JDK16+ 
(just not at the same time - in same maven process when it is forking surefire to different JVM types)

Reply:
@ropalka Can this part be a separate PR? This part has ongoing discussion, and I don't want to delay merging the other part of this while that discussion goes on. I'd like to get the other part and the in-process WF Core upgrade merged this afternoon so we have everything in to get useful results from https://ci.wildfly.org/viewType.html?buildTypeId=WF_WildFlyPreviewLinuxJdk17

An experimental run against an earlier version of this PR showed that job gives useful results without this bit.

https://ci.wildfly.org/viewLog.html?buildId=256802&buildTypeId=WF_WildFlyPreviewLinuxJdk17

Quite a lot of those 78 failures will go away with the core upgrade. Some others are due to the known issue of ByteBuddy needing an upgrade. So I expect we'll be down to a small set.